src/macros.rs

```diff
@@ -249,14 +249,28 @@ macro_rules! __log_module_path {
 #[macro_export]
 macro_rules! __log_file {
     () => {
-        file!()
+        #[cfg(feature = "track_caller")]
+        {
+            ::std::panic::Location::caller().file()
```

Comment:
This cfg is evaluated in the context of the crate that invokes a log macro, not the context of the log crate so this won't work.

Reply:
thanks @sfackler, I pushed an update so they're evaluated in the correct context